### PR TITLE
WooCommerce Docs: Update manifest hash on content update

### PR DESCRIPTION
**Pull request by psealock**
### Submission Review Guidelines:

-   I have followed the [WooCommerce Contributing Guidelines](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md) and the [WordPress Coding Standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/).
-   I have checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change.
-   I have reviewed my code for [security best practices](https://developer.wordpress.org/apis/security/).
-   Following the above guidelines will result in quick merges and clear and detailed feedback when appropriate.

### Changes proposed in this Pull Request:

A new manifest hash needs to be created when content changes. A new hash tells the ingestion portion of WooCommerce Docs to update or create posts. This PR adds a new has for posts based on content. When the content changes, a new hash is created, which causes the manifest hash to update as well.

### How to test the changes in this Pull Request:

Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. Pull down this branch locally.
2. Create a new manifest.
```
pnpm utils md-docs create ./plugins/woocommerce-docs/example-docs woodocs --outputFilePath ./plugins/woocommerce-docs/scripts/manifest.json
```
3. See that no changes have been made to the manifest.
4. Edit any of the markdown files in `/plugins/woocommerce-docs/example-docs`.
5. Recreate the manifest
```
pnpm utils md-docs create ./plugins/woocommerce-docs/example-docs woodocs --outputFilePath ./plugins/woocommerce-docs/scripts/manifest.json
```
6. Confirm the hash from the corresponding post you've edited is different and the root hash of the manifest is also different.
7. Run JS unit tests.

### Changelog entry

-   [ ] Automatically create a changelog entry from the details below.

<details>

#### Significance

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Comment 

</details>

### CODE REVIEW

File: tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts [hunk 1]
```diff
@@ -137,6 +139,104 @@ describe( 'generateManifest', () => {
...
+		// Generate a new manifest
+		const nextManifest = await generateManifestFromDirectory(
+			rootDir,
+			dir,
+			'example-docs',
+			'https://example.com',
+			'https://example.com/edit'
+		);
+		const nextPost = nextManifest.categories[ 0 ].posts[ 0 ];
+		const NextPostHash = nextPost.hash;
```
Comment (lsinger) [inline tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts]: Unclear why this and the next `const` would start with a capital letter? Is there a reason for this that I'm maybe not aware of?

File: tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts [hunk 2]
```diff
@@ -137,6 +139,104 @@ describe( 'generateManifest', () => {
...
```
Comment (lsinger) [inline tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts]: See above.

**Review by lsinger — CHANGES_REQUESTED** [review]
> 3. See that no changes have been made to the manifest.

This changes all IDs and removes all post hashes for me. Consequently changes to a post are not being picked up in the manifest, as there's no hash in the file anymore. The only hash in the file is the manifest one, and that does change when I change a `post_title`, for example. Not sure why the `post.hash` is not being serialized into the manifest for me? 

The unit tests run fine, though. I'm confused as to why they're not catching this behavior. 

Heads up that `.phpunit.result.cache` is not being ignored yet, but I think that might be included in a different PR.

File: tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts (hunk 1 above)
Comment (psealock) [reply tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts]: Fixed in https://github.com/woocommerce/woocommerce/pull/39445/commits/891681e89b5d802769f6770744cbc85cbba328c1

File: tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts (hunk 2 above)
Comment (psealock) [reply tools/monorepo-utils/src/md-docs/lib/__tests__/generate-manifest.ts]: Fixed in https://github.com/woocommerce/woocommerce/pull/39445/commits/891681e89b5d802769f6770744cbc85cbba328c1

**Comment by psealock** [comment]
>This changes all IDs and removes all post hashes for me. Consequently changes to a post are not being picked up in the manifest, as there's no hash in the file anymore

@lsinger The missing hash leads me to believe the changes weren't built. Try this:

1. `pnpm install` will build the Monorepo Utils, or just `cd tools/monorepo-utils && pnpm build`
2. Sometimes I've experienced the build not pick up changes and I need to delete `tools/monorepo-utils/dist` and then rebuild.

**Review by lsinger — APPROVED** [review]
That did it. LGTM! 🚢

**Review by samueljseay — APPROVED** [review]
👏🏻 great stuff, thanks for implementing this.

**Pull request merged**